apps/accelerate/chatllama/chatllama/llama_model.py

```diff
@@ -40,16 +97,314 @@ def __call__(self, texts: Union[List[str], str], *args, **kwargs):
             )
             for i, text in enumerate(texts):
                 tokens[i, : len(text)] = torch.tensor(text).cuda().long()
+
+            # create left padded tokens
+            # TODO: decide how oes and bos should be handled
+            left_tokens = self.tokenizer.pad_id * torch.ones_like(tokens)
+            left_tokens[:, -1] = self.tokenizer.eos_id
+            mask = self.create_sequence_mask(tokens)
+            for i in range(tokens.shape[0]):
+                current_tokens = tokens[i, mask[i] == 1]
+                left_tokens[
+                    i, -len(current_tokens) :  # noqa E203
```

Comment:
shouldn't this be something like 

```suggestion
                    i, -len(current_tokens) - 1 :  -1
```
so that we do not overwrite the `eos_id` token.

Reply:
yep originally was exactly like this, to preserve the final EOS token.
i did change it while debugging and never revert it back. 
Anyway the fact that all the sequences must end with EOS is just assumed by their code, isn't it strange?
I mean if it has to generate the next token (the answer) why do you want a EOS in between?